CMakeLists.txt

```diff
@@ -1,17 +1,54 @@
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 2.8.11)
 
 project(Zopfli)
 
-option(BUILD_SHARED_LIBS "Build Zopfli with shared libraries" OFF)
+# Check if being built standalone
+set(zopfli_standalone FALSE)
+get_directory_property(zopfli_parent_directory PARENT_DIRECTORY)
+if(zopfli_parent_directory STREQUAL "")
+  set(zopfli_standalone TRUE)
+endif()
+unset(zopfli_parent_directory)
+
+#
+# Options
+#
 
-if(NOT CMAKE_BUILD_TYPE)
-  set(CMAKE_BUILD_TYPE Release)
+# Default to BUILD_SHARED_LIBS if set, but allow overriding it
+set(zopfli_shared_default OFF)
+if(DEFINED BUILD_SHARED_LIBS)
+  set(zopfli_shared_default ${BUILD_SHARED_LIBS})
 endif()
+option(ZOPFLI_BUILD_SHARED "Build Zopfli with shared libraries" ${zopfli_shared_default})
+unset(zopfli_shared_default)
 
-if(MSVC)
-  add_definitions(/D_CRT_SECURE_NO_WARNINGS)
+# If standalone or shared subproject, default to building install
+if(zopfli_standalone OR ZOPFLI_BUILD_SHARED)
```

Comment:
I'm unsure why that guard was chosen for enabling the install feature. I'd have expected something like "enabled unless on WIN32" or so. What is the idea?

Reply:
If we are building Zopfli only (standalone), we want to enable install.

If Zopfli is being built as a subproject of some other project, and as static libraries, then we will most likely be linked into the parent project, and should not add our installation files to the parent project (unless asked to).

If however Zopfli is being built as a subproject as shared libraries, it will most likely need to be installed alongside the parent project for it to work.

I don't know, perhaps this is too much logic to put into the CMake script to try and cover obscure use cases?

Reply:
Make more sense to me now, thanks.